tests/integration/test_db.py

```diff
@@ -85,148 +88,97 @@ async def test_finos_waltz_db(ops_test: OpsTest) -> None:
-        )
+    assert leader_unit.workload_status_message == EXTENSIONS_BLOCKING_MESSAGE
 
-        # Verify that active status is restored when all blocking relations are gone
-        await ops_test.model.wait_for_idle(
-            apps=[database_application_name],
-            status="active",
-            raise_on_blocked=False,
-            timeout=1000,
-        )
+    logger.info("Verify that active status is restored when all blocking relations are gone")
+    await ops_test.model.applications[DATABASE_APP_NAME].destroy_relation(
+        f"{DATABASE_APP_NAME}:db", f"{APPLICATION_NAME}2:db"
+    )
 
-        # Verify that the charm doesn't block when the extensions are enabled.
-        logger.info("Verifying that the charm doesn't block when the extensions are enabled")
-        config = {"plugin_pg_trgm_enable": "True", "plugin_unaccent_enable": "True"}
-        await ops_test.model.applications[database_application_name].set_config(config)
-        await ops_test.model.wait_for_idle(
-            apps=[database_application_name], status="active", idle_period=15
-        )
-        await ops_test.model.relate(f"{database_application_name}:db", "indico1:db")
-        await ops_test.model.wait_for_idle(
-            apps=[database_application_name, "indico1"],
-            status="active",
-            raise_on_blocked=False,
-            timeout=2000,
-        )
+    await ops_test.model.wait_for_idle(
+        apps=[DATABASE_APP_NAME],
+        status="active",
+        raise_on_blocked=False,
+        timeout=1000,
+    )
 
-        # Verify that the charm unblocks when the extensions are enabled after being blocked
-        # due to disabled extensions.
-        logger.info("Verifying that the charm unblocks when the extensions are enabled")
-        config = {"plugin_pg_trgm_enable": "False", "plugin_unaccent_enable": "False"}
-        await ops_test.model.applications[database_application_name].set_config(config)
-        await ops_test.model.applications[database_application_name].destroy_relation(
-            f"{database_application_name}:db", "indico1:db"
-        )
-        wait_for_relation_removed_between(ops_test, database_application_name, "indico1")
-        await ops_test.model.wait_for_idle(
-            apps=[database_application_name, "indico1"], status="active", idle_period=15
-        )
+    logger.info("Verifying that the charm doesn't block when the extensions are enabled")
+    config = {"plugin_pg_trgm_enable": "True", "plugin_unaccent_enable": "True"}
+    await ops_test.model.applications[DATABASE_APP_NAME].set_config(config)
+    await ops_test.model.wait_for_idle(apps=[DATABASE_APP_NAME], status="active")
+    await ops_test.model.relate(f"{DATABASE_APP_NAME}:db", f"{APPLICATION_NAME}:db")
+    await ops_test.model.wait_for_idle(
+        apps=[DATABASE_APP_NAME, APPLICATION_NAME],
+        status="active",
+        raise_on_blocked=False,
+        timeout=2000,
```

Comment:
Can we change `raise_on_blocked` to `True` here and in the other places where the charm shouldn't block? It can speedup the test if something is wrong.

Reply:
I think we don't really need it in any of the `wait_for_idle` calls (we track it with `block_until`).